libtransmission/utils.c

```diff
@@ -403,6 +403,21 @@ int64_t tr_getDirFreeSpace(char const* dir)
     return free_space;
 }
 
+bool tr_canWriteFile(char const* path)
+{
+    FILE* file = fopen(path, "wb+");
```

Comment:
Don't use neither `open` nor `fopen`, but instead our [Unicode-aware] filesystem wrappers in libtransmission/file.h (already included here). This is because `open` is non-portable (and you already fixed that) and `fopen` doesn't support UTF-8 (which all paths in Transmission are expected to be in) on Windows.

If file.h doesn't provide the necessary functionality, it'll have to be added. For instance, I'm not convinced that opening the file to check that the location is writable and whether the file exists is better than using `access(path, W_OK)` (which doesn't leave the file behind); the latter seems more suitable for a general-purpose helper function which this one tends to be.

Reply:
I'll have a look at all this over the weekend thanks for your help.